### PR TITLE
[Vertex AI] Migrate logging to `FirebaseLogger`

### PR DESCRIPTION
**Pull request by andrewheard**
Migrated the Vertex AI in Firebase SDK to use the standard `FirebaseLogger` implementation instead of direct calls to Swift Logger.

#no-changelog

### CODE REVIEW

File: FirebaseVertexAI/Sources/VertexLog.swift [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+import Foundation
+import os.log
+
+@_implementationOnly import FirebaseCoreExtension
+
+enum VertexLog {
+  /// Log message codes for the Vertex AI SDK
+  ///
+  /// These codes should ideally not be re-used in order to facillitate matching error codes in
```
Comment (paulb777) [inline FirebaseVertexAI/Sources/VertexLog.swift:23]: typo: facilitate

File: FirebaseVertexAI/Sources/VertexLog.swift [hunk 2]
```diff
@@ -0,0 +1,112 @@
+// Copyright 2023 Google LLC
```
Comment (paulb777) [inline FirebaseVertexAI/Sources/VertexLog.swift:1]: Did this come from somewhere? Otherwise 2024

**Review by paulb777 — APPROVED** [review]

File: FirebaseVertexAI/Sources/VertexLog.swift (hunk 1 above)
Comment (andrewheard) [reply FirebaseVertexAI/Sources/VertexLog.swift]: Thanks for catching. Fixed.

File: FirebaseVertexAI/Sources/VertexLog.swift (hunk 2 above)
Comment (andrewheard) [reply FirebaseVertexAI/Sources/VertexLog.swift]: Fixed to 2024.

**Pull request merged**